src/main/java/uk/co/bconline/ndelius/service/impl/UserServiceImpl.java

```diff
@@ -109,6 +109,8 @@ public UserServiceImpl(
 			var stream = allOf(groupMembersFuture, ldapFuture, dbFuture)
 					.thenApply(v -> Stream.of(ldapFuture.join(), dbFuture.join())).join()
 					.flatMap(Collection::stream)
+					.map(sr -> expandEmailSearchToDB(sr, query))
+					.flatMap(Collection::stream)
```

Comment:
```suggestion
					.flatMap(sr -> expandEmailSearchToDB(sr, query))
```
You can combine these lines if you update `expandEmailSearchToDB` to return a Stream instead of a List.